k8s/cmd/admission-webhook/patches.go

```diff
@@ -95,10 +95,17 @@ func createDNSPatch(tuple *podSpecAndMeta, annotationValue string) (patch []patc
 func createNsmInitContainerPatch(annotationValue string) []patchOperation {
 	var patch []patchOperation
 
-	envVals := []corev1.EnvVar{{
-		Name:  client.AnnotationEnv,
-		Value: annotationValue,
-	}}
+	namespace := getNamespace()
+	envVals := []corev1.EnvVar{
+		{
+			Name:  client.AnnotationEnv,
+			Value: annotationValue,
+		},
+		{
+			Name:  client.NamespaceEnv,
+			Value: namespace,
+		},
+	}
```

Comment:
Is this needed?

Reply:
This is a fix for the proper k8s namespace to be exposed in crossconnect-monitor. Previously it was always `default` for the client.